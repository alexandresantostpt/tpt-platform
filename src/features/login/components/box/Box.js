import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

import { borders } from '@helpers/borders'

const StyledBox = styled.div`
    background: #fdfdfd;
    border: ${borders.default};
    border-radius: 10px;
    box-shadow: 5px 5px 10px 0 rgba(0, 0, 0, 0.2);
    opacity: 0.85;
    padding: 2.2rem 1.75rem;
`

const Box = ({ children }) => <StyledBox>{children}</StyledBox>

Box.propTypes = {
    children: PropTypes.oneOfType([PropTypes.array, PropTypes.func, PropTypes.object, PropTypes.string]).isRequired
}

export default Box
