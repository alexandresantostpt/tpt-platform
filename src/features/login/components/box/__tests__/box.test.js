import React from 'react'
import renderer from 'react-test-renderer'

import Box from '../Box'

describe('Tests for Box component', () => {
    it('Should rendering without crash', () => {
        const tree = renderer
            .create(
                <Box>
                    <h1>Must be show!</h1>
                </Box>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
