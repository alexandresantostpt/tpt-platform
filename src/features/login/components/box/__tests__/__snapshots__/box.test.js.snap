// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Box component Should rendering without crash 1`] = `
<div
  className="Box__StyledBox-v5mj6b-0 llPnpv"
>
  <h1>
    Must be show!
  </h1>
</div>
`;
